src/main/java/com/google/step/model/User.java

```diff
@@ -23,7 +23,11 @@ public User(long id, String email, String username, String photoBlobKey, String
     this.username = username;
     this.email = email;
     this.bio = bio;
-    this.photoBlobKey = Optional.of(photoBlobKey);
+    if (photoBlobKey != null) {
+      this.photoBlobKey = Optional.of(photoBlobKey);
+    } else {
+      this.photoBlobKey = null;
+    }
```

Comment:
It feels weird to have three states - `null`, `Optional.absent()` and `Optional.of("some value")`.

Maybe you should just use `Optional.fromNullable()`, which will use `Optional.absent()` for null values?

Reply:
In the update user method in user data manager, I pass in a User object with some null fields to update non-null filed and keep other null field unchanged.